website/content/en/docs/upgrading/v1-migration.md

```diff
@@ -176,233 +289,297 @@ kubectl patch customresourcedefinitions ec2nodeclasses.karpenter.k8s.aws -p "{\"
+        kind: EC2NodeClass
+        name: nodeclass
+---
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+metadata:
+  name: nodeclass
 ```
 
-4. Rollback the CRDs
-
-```bash
-helm upgrade --install karpenter-crd oci://public.ecr.aws/karpenter/karpenter-crd --version "${KARPENTER_VERSION}" --namespace "${KARPENTER_NAMESPACE}" --create-namespace \
-  --set webhook.enabled=true \
-  --set webhook.serviceName=karpenter \
-  --set webhook.port=8443
+In this example, we have two `NodePools` with different `kubelet` values, but they refer to the same `EC2NodeClass`.
+When handling conversion, Karpenter will annotate the `NodePools` with the `compatibility.karpenter.sh/v1beta1-kubelet-conversion` annotation.
+Before upgrading to `v1.1`, you must update your `NodePools` to refer to separate `EC2NodeClasses` to retain this behavior.
+
+```yaml
+apiVersion: karpenter.sh/v1
+kind: NodePool
+metadata:
+  name: nodepool-a
+spec:
+  template:
+    spec:
+      nodeClassRef:
+        group: karpenter.k8s.aws
+        kind: EC2NodeClass
+        name: nodeclass-a
+---
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: nodepool-b
+spec:
+  template:
+    spec:
+      nodeClassRef:
+        group: karpenter.k8s.aws
+        kind: EC2NodeClass
+        name: nodeclass-b
+---
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+metadata:
+  name: nodeclass-a
+spec:
+  kubelet:
+    maxPods: 10
+---
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+metadata:
+  name: nodeclass-b
+spec:
+  kubelet:
+    maxPods: 20
 ```
 
-{{% alert title="Note" color="warning" %}}
-
-As an alternative approach to updating the Karpenter CRDs conversion webhook configuration, you can patch the CRDs as follows:
-
-```bash
-export SERVICE_NAME=<karpenter webhook service name>
-export SERVICE_NAMESPACE=<karpenter webhook service namespace>
-export SERVICE_PORT=<karpenter webhook service port>
-# NodePools
-kubectl patch customresourcedefinitions nodepools.karpenter.sh -p "{\"spec\":{\"conversion\":{\"webhook\":{\"clientConfig\":{\"service\": {\"name\": \"${SERVICE_NAME}\", \"namespace\": \"${SERVICE_NAMESPACE}\", \"port\":${SERVICE_PORT}}}}}}}"
-# NodeClaims
-kubectl patch customresourcedefinitions nodeclaims.karpenter.sh -p "{\"spec\":{\"conversion\":{\"webhook\":{\"clientConfig\":{\"service\": {\"name\": \"${SERVICE_NAME}\", \"namespace\": \"${SERVICE_NAMESPACE}\", \"port\":${SERVICE_PORT}}}}}}}"
-# EC2NodeClass
-kubectl patch customresourcedefinitions ec2nodeclasses.karpenter.k8s.aws -p "{\"spec\":{\"conversion\":{\"webhook\":{\"clientConfig\":{\"service\": {\"name\": \"${SERVICE_NAME}\", \"namespace\": \"${SERVICE_NAMESPACE}\", \"port\":${SERVICE_PORT}}}}}}}"
-```
+{{% alert title="Note" color="primary" %}}
+Updating the `nodeClassRef` for your `NodePools` will cause those `NodePools`' nodes to drift.
 {{% /alert %}}
 
-5. Rollback the Karpenter Controller
+<!-- TODO (jmdeal@): Add a section on storage version migration -->
```

Comment:
Will this be going in the PR or another one? Since we have already released seem like we should add it now, no?